.github/workflows/pr-build.yml

```diff
@@ -0,0 +1,51 @@
+name: Build pull request
+
+on:
+  pull_request:
+    paths:
+      - ".github/workflows/pr-build.yml"
+      - "app/**"
+      - "gradle/**"
+      - "build.gradle.kts"
+      - "gradle.properties"
+      - "settings.gradle.kts"
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set env
+        run: echo "COMMIT_HASH=$(git rev-parse --short HEAD)" >> $GITHUB_ENV
+
+      - name: Set up cache
+        uses: actions/cache@v3
+        with:
+          path: |
+            ${{ runner.home }}/.gradle/caches
+            ${{ runner.home }}/.gradle/wrapper
+            .gradle
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+
+      - name: Set up Java
+        run: echo "JAVA_HOME=$JAVA_HOME_17_X64" >> $GITHUB_ENV
+
+      - name: Generate keystore
+        run: keytool -genkey -v -keystore debugkey.jks -alias androiddebugkey -storepass android -keypass android -keyalg RSA -keysize 2048 -validity 10000 -dname "O=ReVanced"
+
+      - name: Build with Gradle
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: ./gradlew assembleRelease --no-daemon -Pandroid.injected.signing.store.file=$GITHUB_WORKSPACE/debugkey.jks -Pandroid.injected.signing.store.password=android -Pandroid.injected.signing.key.alias=androiddebugkey -Pandroid.injected.signing.key.password=android
+
+      - name: Add hash to APK
+        run: mv app/build/outputs/apk/release/app-release.apk revanced-manager-${{ env.COMMIT_HASH }}.apk
```

Comment:
These steps should ideally be included in the build system. Depending on the type of build, it could append the hash to the file name. 

Here is an example: https://github.com/iBotPeaches/Apktool/blob/master/build.gradle.kts#L50

Let me know what you think about this.

Reply:
Sounds good. It would clean things up a bit.

Reply:
By the way, what do you think about having pr builds having minify disabled.

Reply:
Just to clarify, would I be adding a new assemble task to gradle such as `assembleReleaseTesting`?

Reply:
I am not sure. I think it is possible to make a task depend on another. So you can make the assembleRelease task depend on the renaming task maybe?

Reply:
Oh I get now. Cool.

Reply:
Should it only rename when given a keystore? Currently it outputs as `app-release-unsigned` with no key and `app-release` with a key?

Reply:
So when given no key it should output the same thing but with `-unsigned` at the end?

Reply:
yep!

Reply:
I think it doesn't matter too much. What I sent is just for reference. You can append `-unsigned` at the end. I think @validcube already renames the file to begin with if its a debug build.

Reply:
Should it just append the git hash at the end?

Reply:
> I think that is possible. A release task can sign a release apk, but the hash should not be added. The release task only runs when a release is made, which happens in main. The release file name is appended with the tag assigned to HEAD.

When would a hash be added to the file name then? If the release task does not add it.

Reply:
> When you locally debug, you should probably rely on a debug release task so that you can actually debug

This is true for most cases, except when testing performance

Reply:
> You could configure only minify if for example proguard was in the gradle start properties

```
buildTypes {
        release {
            if ("proguard" in gradle.startParameter.taskNames) {
                isMinifyEnabled = true
                isShrinkResources = true
                proguardFiles(getDefaultProguardFile("proguard-android-optimize.txt"), "proguard-rules.pro")
            }
        }
    }
```

Reply:
In this case two tasks can be made or the task can be made configurable to build releases or debug releases.

Reply:
> When would a hash be added to the file name then? If the release task does not add it.

A separate task made for PR builds I suppose.

Reply:
> A separate task made for PR builds I suppose.

With having a configurable assembleRelease task, it could normally always append the hash, but when adding the extra param, it could minify AND rename with tag instead of hash.

What to call that param though, `assembleRelease debug` doesn't seem right...

Reply:
How about making `assembleRelease` work as it does right now, and when using the parameter `debug`, append the has, sign with a debug key as well as don't minify?

Reply:
I am not sure if I understood correctly. The task `assembleRelease` should work as it works right now, andwhen using the parameter `debug`, the hash can be appended, sig with a debug key as well as don't minify

Reply:
Yesn

Reply:
I was meaning if we had a task called `installRelease` to install the release build to a device, it should run the `assembleRelease` task with the `debug` param since otherwise it would be trying to install an unsigned apk.

Reply:
@oSumAtrIX Ok the changes have been done:

- `assembleRelease` will output `revanced-manager-release-unsigned.apk`
- `assembleRelease -Pdebug` will output `revanced-manager-{git hash}-release.apk` (this build will also have minify disabled)
- `installRelease` will fail as you cannot install an unsigned apk
- `installRelease -Pdebug` will install the signed release build
- `assembleDebug` will output `revanced-manager-debug,apk`

Also let me know what you think about the prefix change from `app` to `revanced-manager`.

Reply:
@Ushie Is the `assembleRelease` fine? How would the tag be appended when pushing a tag on main?

Reply:
It could either be renamed manually or I could add a param like `-PaddTag` to rename it

Reply:
it would probably be used as `-Ptag=${GITHUB_REF#refs/*/}` if you want to use a param

Reply:
> @Ushie Is the `assembleRelease` fine? How would the tag be appended when pushing a tag on main?

```yaml
- name: Set env
  run: echo "RELEASE_VERSION=${GITHUB_REF#refs/*/}" >> $GITHUB_ENV
```

This is what's currently done in https://github.com/ReVanced/revanced-manager/blob/main/.github/workflows/release-build.yml

Reply:
Updated params:

- `assembleRelease` will output `app-release-unsigned.apk`
- `assembleRelease -Psign` will output `app-release.apk`
- `assembleRelease -PnoProguard` will disable proguard
- `installRelease` will fail as you cannot install an unsigned apk
- `installRelease -Psign` will install the signed release build
- `installRelease -PnoProguard -Psign` will install the signed release build without proguard